Add resource requests/limits for KEDA deployment

JorTurFer opened the pull request:
This PR adds default values in chart for resources. This default values are the [values provided in the documentation](https://keda.sh/docs/2.4/operate/cluster/#cluster-capacity-requirements)

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/master/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*
- [x] README is updated with new configuration values *(if applicable)*

Fixes https://github.com/kedacore/charts/issues/112

JorTurFer commented:
sure :)

JorTurFer commented:
One more thing that comes to my mind...
If it's already "planned" a [huge refactor in the values structure](https://github.com/kedacore/charts/issues/182), does it make sense to split in resources section in 2 right now? 
I mean, this is a breaking change, and the other will be another breaking change, maybe it is better just to set the default values in this PR, and split in two inside the scope of the refactor
wdyt?

tomkerkhove commented:
> maybe it is better just to set the default values in this PR, and split in two inside the scope of the refactor (we avoid one breaking change)

Fine by me as long as you drop a note in there so that we don't forget 😅

JorTurFer commented:
hi @tomkerkhove 
I have reverted all the changes in favor of a simple definition of resources and I added a comment in this PR and also in the refactor's issue to don't forget this pending change

tomkerkhove commented:
Thanks!

JorTurFer commented:
You're welcome ^^
